janus_server/src/binary_utils.rs

```diff
@@ -60,3 +70,149 @@ pub fn datastore<C: Clock>(
+    )]
+    otlp_tracing_metadata: Vec<(String, String)>,
+
+    /// Additional OTLP/gRPC metadata key/value pairs. (concatenated with those in the metrics
+    /// configuration sections)
+    #[structopt(
+        long,
+        env = "OTLP_METRICS_METADATA",
+        parse(try_from_str = parse_metadata_entry),
+        help = "additional OTLP/gRPC metadata key/value pairs for the metrics exporter",
+        value_name = "KEY=value",
+        use_delimiter = true,
+    )]
+    otlp_metrics_metadata: Vec<(String, String)>,
+}
+
+impl Debug for CommonBinaryOptions {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Options")
+            .field("config_file", &self.config_file)
+            .finish()
+    }
+}
+
+fn parse_metadata_entry(input: &str) -> Result<(String, String)> {
+    if let Some(equals) = input.find('=') {
+        let (key, rest) = input.split_at(equals);
+        let value = &rest[1..];
+        Ok((key.to_string(), value.to_string()))
+    } else {
+        Err(anyhow!(
+            "`--otlp-tracing-metadata` and `--otlp-metrics-metadata` must be provided a key and value, joined with an `=`"
+        ))
+    }
+}
+
+pub async fn janus_main<O, C, Config, F, Fut>(clock: C, f: F) -> anyhow::Result<()>
+where
+    O: BinaryOptions,
+    C: Clock,
+    Config: BinaryConfig,
+    F: FnOnce(C, Config, Datastore<C>) -> Fut,
+    Fut: Future<Output = anyhow::Result<()>>,
+{
+    // Read arguments, read & parse config.
+    let options = O::from_args();
+    let common_options = options.common_options();
+    let mut config = {
+        let config_content =
+            fs::read_to_string(&common_options.config_file).with_context(|| {
+                format!("couldn't read config file {:?}", common_options.config_file)
+            })?;
+        let mut config: Config = serde_yaml::from_str(&config_content).with_context(|| {
+            format!(
+                "couldn't parse config file {:?}",
+                common_options.config_file
+            )
+        })?;
+
+        if let Some(OpenTelemetryTraceConfiguration::Otlp(otlp_config)) =
+            &mut config.logging_config().open_telemetry_config
+        {
+            otlp_config
+                .metadata
+                .extend(common_options.otlp_tracing_metadata.iter().cloned());
+        }
+        if let Some(MetricsExporterConfiguration::Otlp(otlp_config)) =
+            &mut config.metrics_config().exporter
+        {
+            otlp_config
+                .metadata
+                .extend(common_options.otlp_metrics_metadata.iter().cloned());
+        }
+
+        config
+    };
+    install_trace_subscriber(config.logging_config())
+        .context("couldn't install tracing subscriber")?;
+    let _metrics_exporter = install_metrics_exporter(config.metrics_config())
+        .context("failed to install metrics exporter")?;
+
+    info!(?common_options, ?config, "Starting aggregation job creator");
```

Comment:
nit: this info line refers to the aggregation job creator specifically; I suspect we can get away without referencing which job is starting up, so the easiest fix might be to just change the string to something like `"Starting up"`.

Reply:
Yeah, I think in practice it'll be obvious which binary logged this because the log event will be annotated with a Kubernetes pod or deployment.